### PR TITLE
Fix --tail log on restart problem

### PR DESCRIPTION
**Pull request by rhatdan**
--tail=1 is not working f you restart a container with journald logging.

We see the exit status and then call into the logging a second time
causing all of the logs to print.

Removing the tail log on exited seems to fix the problem.

Fixes: https://github.com/containers/podman/issues/13098

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

**Comment by nalind** [comment]
If it passes tests and doesn't bring back #12263, it's fine.

File: test/system/035-logs.bats
Comment (edsantiago) [inline test/system/035-logs.bats:34]: unused

File: test/system/035-logs.bats
Comment (edsantiago) [inline test/system/035-logs.bats:45]: Could you rename this message? Maybe "logs, after restart, should still only show last line"? Distinct error messages make it sooooo much easier to track down the exact failure.

**Review by edsantiago — CHANGES_REQUESTED** [review]
Mostly LGTM, with a few nits, and one big question: should the new test conform to the existing convention of testing under both `k8s-file` and `journald`?

Actually, two questions: regardless of the answer to the above, I think `skip_if_journald_unavailable` might be needed to avoid the usual mad panic when this makes it into RHEL-land.

**Comment by edsantiago** [comment]
New push addresses two of my nits. Can you, for posterity, at least comment on my other questions? I don't know where we stand these days on rootless RHEL journald.

**Comment by rhatdan** [comment]
I think we should test with both, if possible.

**Comment by edsantiago** [comment]
I'm interpreting that as a request for code:
```bash
function _log_test_tail() {
    local driver=$1

    run_podman run -d --log-driver=$driver $IMAGE sh -c "echo test1; echo test2"
    cid="$output"

    run_podman logs --tail 1 $cid
    is "$output" "test2"  "logs should only show last line"

    run_podman restart $cid

    run_podman logs --tail 1 $cid
    is "$output" "test2"  "logs should only show last line after restart"

    run_podman rm $cid
}

@test "podman logs - tail test, k8s-file" {
    _log_test_tail k8s-file
}

@test "podman logs - tail test, journald" {
    # We can't use journald on RHEL as rootless: rhbz#1895105
    skip_if_journald_unavailable

    _log_test_tail journald
}
```

**Comment by rhatdan** [comment]
@edsantiago Added your code.

**Comment by edsantiago** [comment]
LGTM assuming tests pass. FYI I ran new tests against `main`, and the `k8s-file` one passes root & rootless, `journald` fails root & rootless, which means both test and code are doing the right thing.

/lgtm
/hold

**Review by edsantiago — APPROVED** [review]

**Pull request merged**
